src/compiler/diagnosticMessages.json

```
@@ -319,6 +319,10 @@
         "category": "Error",
         "code": 1105
     },
+    "The left-hand side of a for-of statement may not be 'async'.": {

```

TIL we actually spell this `'for...of' statement` in other places. I think it should be for-of/for-in, but this PR should be consistent with existing messages, and then we can change the messages all at once if others agree with me.

```suggestion
    "The left-hand side of a 'for...of' statement may not be 'async'.": {
```